### PR TITLE
Console prod issue

### PR DESCRIPTION
**Pull request by TanujaG**
Prod issue for console when deleting of the courses having names with special chars and notifications not rendering as expected with dangerouslySetInnerHTML
Fix:To use dangerouslySetInnerHTML only it the notification body have HTML Tags in it.

@loahou04 @Sauravvaidya @nlkluth

### CODE REVIEW

**Comment by loahou04** [comment]
Looks good to me!

File: src/js/utils.js [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+
+module.exports = {
+	hasHTMLTags : function(s) {
+		const doc = new DOMParser().parseFromString(s, 'text/html');
+		const childNodes = doc && [].slice.call(doc.body.childNodes);
+		return this.hasElementNodeType(childNodes);
+	},
+
+	hasElementNodeType : function (nodes) {	
+		return nodes && nodes.filter((node) => node.nodeType === 1).length > 0;
```
Comment (Sauravvaidya) [inline src/js/utils.js]: Just .length should also do the job :)

File: src/js/NotificationSummary.js [hunk 2]
```diff
@@ -0,0 +1,15 @@
+import React from 'react';
+import utils from './utils';
+
+const NotificationSummary = ({summary, className}) => {
+	if(summary && utils.hasHTMLTags(summary)) {
```
Comment (Sauravvaidya) [inline src/js/NotificationSummary.js]: If the summary has HTML defined and some bad tags, we still will pass this check as some nodes will have nodeType as 1, then we try to render that html, aren't we in the same boat as before?

**Comment by marcdonovan** [comment]
Hi Tanuja

**Comment by loahou04** [comment]
LOL we need to take your access away!!!! lol

**Comment by marcdonovan** [comment]
This is open source dude

**Comment by TanujaG** [comment]
Hi marc :) ... how are you ?

**Comment by marcdonovan** [comment]
doing great, thanks. You?

**Comment by marcdonovan** [comment]
image

**Comment by marcdonovan** [comment]
You’ve been removed from the Origami Component Developers team on the Pearson Higher Ed organization.

so that's how we roll?

**Comment by loahou04** [comment]
Hahah i didn't request or do the removal...must be someone else who did that lol

**Comment by Sauravvaidya** [comment]
Looks good to me. More testing should be done after console brings in this new version and updates their payload. Also existing users with this error should not see the errors anymore.

**Pull request merged**